### PR TITLE
SLLS-122 support QuickFixes with notebooks

### PR DESCRIPTION
**Pull request by sophio-japharidze-sonarsource**

### CODE REVIEW

**Review by kirill-knize-sonarsource — CHANGES_REQUESTED** [review]
Some cleanup needed.

File: src/main/java/org/sonarsource/sonarlint/ls/AnalysisScheduler.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ public void didClose(URI fileUri) {
    */
   Future<?> analyzeAsync(List<VersionedOpenFile> files, boolean shouldFetchServerIssues) {
     var trueFileUris = files.stream().filter(f -> {
-      if (!Utils.uriHasFileSchema(f.getUri())) {
+      if (!Utils.uriHasFileScheme(f.getUri())) {
```
[the remark this answers is not in the document]
Comment (jblievremont) [reply src/main/java/org/sonarsource/sonarlint/ls/AnalysisScheduler.java:174]: Pretty sure it's [scheme](https://en.wikipedia.org/wiki/List_of_URI_schemes) 😇

**Review by jblievremont — CHANGES_REQUESTED** [review]
From a pure functional point of view this seems to do the job.

I believe the code would be more robust and future-proof if it relied on the LSP-backed relationship between notebooks and cells, rather on what looks like VSCode-specific implementation details.

**Review by kirill-knize-sonarsource — APPROVED** [review]

**Review by jblievremont — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**